corehq/apps/userreports/models.py

```diff
@@ -755,12 +758,35 @@ def rebuild_failed(self):
     def rebuild_awaiting_or_in_progress(self):
         return self.meta.build.awaiting or (self.meta.build.is_rebuild_in_progress and not self.rebuild_failed)
 
+    def set_rebuild_flags(self):
+        from .rebuild import get_table_diffs
+
+        if not self.rebuild_awaiting_or_in_progress:
+            engine = connection_manager.get_engine(self.engine_id)
+            table_name = get_table_name(self.domain, self.table_id)
+            engine_metadata = get_metadata(self.engine_id)
+            diffs = get_table_diffs(engine, [table_name], engine_metadata)
+            tables_to_rebuild = {
+                diff.table_name
+                for diff in diffs
+                if diff.type in DiffTypes.TYPES_FOR_REBUILD
+            }
```

Comment:
nit: I assume this is duplicated, so can it be extracted.
Fine to ignore if not duplicated.

Reply:
Yeah, there is a function that returns the same dictionary. 0f8db3b979e

Reply:
okay, i thought the whole loop was same as in pillows.